lib/posix/pthread.c

```diff
@@ -35,7 +35,8 @@ static const pthread_attr_t init_pthread_attrs = {
 };
 
 static struct posix_thread posix_thread_pool[CONFIG_MAX_PTHREAD_COUNT];
-static u32_t pthread_num;
+static int posix_thread_pool_usage[CONFIG_MAX_PTHREAD_COUNT];
```

Comment:
Can we use a bitfield instead and save 32x memory?

see thread_idx_alloc() and thread_idx_free() in kernel/userspace.c for an example